org-edit-latex.el

```diff
@@ -297,6 +297,20 @@ latex-environment."
           (apply oldfun args)))
     (apply oldfun args)))
 
+;;;###autoload
+(defun org-edit-latex-smart-hint ()
+  (if (and (equal major-mode 'org-mode)
+           (or (eq 'latex-environment (car (org-element-context)))
+               (org-inside-LaTeX-fragment-p)))
```

Comment:
`org-inside-LaTeX-fragment`, which is designed to be used with cdlatex. returns a non-nil value whenever the cursor is behind a `$`, `\(` or `\[`, even it is not closed yet.  Unlike cdlatex, `org-edit-latex` depends on `org-element` api heavily.

In short, this line could cause an issue. On the other hand, line 303 looks good.